common/qodana.ts

```diff
@@ -24,39 +26,36 @@ export function getQodanaSha256(archiveName: string): string {
     case 'darwin_arm64':
       return 'f69c832feb2c223bfa209fc38262e8559e049dca48ca7b387f7aa19b8fa6d637'
     default:
-      throw new Error(`Qodana CLI does not exist for ${archiveName}`)
+      throw new Error(`Qodana CLI does not exist for ${platform}_${arch}`)
   }
 }
 
 /**
- * Returns the message when Qodana binary is corrupted.
- * @param expected expected sha256 checksum
- * @param actual actual sha256 checksum
+ * Returns the architecture name suitable for the published Qodana CLI archive name.
  */
-export function getQodanaSha256MismatchMessage(
-  expected: string,
-  actual: string
-): string {
-  return `Downloaded Qodana CLI binary is corrupted. Expected SHA-256 checksum: ${expected}, actual checksum: ${actual}`
+export function getProcessArchName(): string {
+  return process.arch === 'x64' ? 'x86_64' : 'arm64'
 }
 
-export function getQodanaArchiveName(arch = '', platform = ''): string {
-  if (arch === '') {
-    arch = process.arch === 'x64' ? 'x86_64' : 'arm64'
-  }
-  if (platform === '') {
-    platform = process.platform
+/**
+ * Returns the platform name suitable for the published Qodana CLI archive name.
+ */
+export function getProcessPlatformName(): string {
+  return process.platform === 'win32' ? 'windows' : process.platform
+}
+
+/**
+ * Gets Qodana CLI download URL from the GitHub Releases API.
+ */
+export function getQodanaUrl(arch: string, platform: string): string {
```

Comment:
I think it would be better to type "arch" and "platform" args 

```ts
export type Platforms = 'arm64' | 'x86_64'

export type Arches = 'win32' | 'linux' | 'darwin'

function foo (platform: Platforms, arch: Arches) {

}
```